cloudsmith_cli/cli/commands/repos.py

```diff
@@ -0,0 +1,308 @@
+        utils.pretty_print_table(headers, rows)
+
+    click.echo()
+
+    num_results = len(data)
+    list_suffix = "repositor%s visible" % ("ies" if num_results != 1 else "y")
+    utils.pretty_print_list_info(
+        num_results=num_results, page_info=page_info, suffix=list_suffix
+    )
+
+
+@main.group(cls=command.AliasGroup, name="repositories", aliases=["repos"])
+@decorators.common_cli_config_options
+@decorators.common_cli_output_options
+@decorators.common_api_auth_options
+@decorators.initialise_api
+@click.pass_context
+def repositories(ctx, opts):  # pylink: disable=unused-argument
+    """
+    Manage Repositories.
+
+    See the help for subommands for more information on each.
+    """
+
+
+@repositories.command(name="get", aliases=["list", "ls"])
+@decorators.common_cli_config_options
+@decorators.common_cli_list_options
+@decorators.common_cli_output_options
+@decorators.common_api_auth_options
+@decorators.initialise_api
+@click.argument(
+    "owner_repo",
+    metavar="OWNER/REPO",
+    callback=validators.validate_optional_owner_repo,
+    default="",
+    required=False,
+)
+@click.pass_context
+def get(ctx, opts, owner_repo, page, page_size):
+    """
+    List repositories for a namespace (owner).
+
+    OWNER/REPO: Specify the OWNER namespace (i.e user or org) to list the
+    repositories for that namespace.
+
+    If REPO isn't specified, all repositories will be retrieved from the
+    OWNER namespace.
+
+    If OWNER isn't specified it'll default to the currently authenticated user
+    (if any). If you're unauthenticated, no results will be returned.
+    """
+    # Use stderr for messages if the output is something else (e.g.  # JSON)
+    use_stderr = opts.output != "pretty"
+
+    click.echo("Getting list of repositories ... ", nl=False, err=use_stderr)
+
+    if isinstance(owner_repo, list):
+        if len(owner_repo) == 1:
+            owner = owner_repo[0]
+            repo = None
+        else:
+            owner, repo = owner_repo
+    if isinstance(owner_repo, str):
+        repo = None
+
+        if owner_repo:
+            owner = owner_repo
+        else:
+            owner = None
+
+    context_msg = "Failed to get list of repositories!"
+    with handle_api_exceptions(ctx, opts=opts, context_msg=context_msg):
+        with maybe_spinner(opts):
+            repos_, page_info = api.list_repos(
+                owner=owner, repo=repo, page=page, page_size=page_size
+            )
+
+    click.secho("OK", fg="green", err=use_stderr)
+
+    if utils.maybe_print_as_json(opts, repos_, page_info):
+        return
+
+    print_repositories(opts=opts, data=repos_, show_list_info=False)
+
+
+@repositories.command(aliases=["new"])
+@decorators.common_cli_config_options
+@decorators.common_cli_output_options
+@decorators.common_api_auth_options
+@decorators.initialise_api
+@click.argument("owner", default=None, required=True)
+@click.argument("repo_config_file", type=click.File("rb"), required=True)
```

Comment:
Why a JSON file rather than just CLI options?

Reply:
It was the option that Lee and I opted for when discussing the ticket, since the commands could get lengthy, especially with a `description` option e.g. `cloudsmith repos create namespace --name name --slug slug --type type --description`

We could support both, but then I guess there's the question of precedence when both are included